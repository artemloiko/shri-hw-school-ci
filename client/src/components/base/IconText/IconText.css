.icon-text {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
}

.icon-text__icon {
  opacity: 0.3;
}

.icon-text__text {
  margin-left: var(--space-xxxs);
  font-size: var(--typo-size-s);
  line-height: 1.25;
  word-break: break-all;
}

.icon-text__secondary-text {
  color: var(--color-text-secondary);
  word-break: normal;
}
