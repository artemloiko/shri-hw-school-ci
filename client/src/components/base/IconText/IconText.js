import React from 'react';
import PropTypes from 'prop-types';
import { cn } from 'utils/bem-cn';

import './IconText.css';

function IconText(props) {
  const { icon, text, secondaryText } = props;

  return (
    <div className={cn('icon-text', props)}>
      {icon}
      <div className="icon-text__text">{text}</div>
      {secondaryText && (
        <div className="icon-text__text icon-text__secondary-text">{secondaryText}</div>
      )}
    </div>
  );
}

IconText.propTypes = {
  icon: PropTypes.element,
  text: PropTypes.string,
  secondaryText: PropTypes.string,
  children: PropTypes.string,
};

export default IconText;
