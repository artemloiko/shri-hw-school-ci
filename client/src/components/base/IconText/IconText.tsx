import React from 'react';
import { cn, CNProps } from '../../../utils/bem-cn';

import Icon, { IconTypes } from '../Icon/Icon';

import './IconText.css';

export type IconTextProps = CNProps & {
  iconType: IconTypes;
  text: string;
  secondaryText?: string;
};

const IconText: React.FC<IconTextProps> = (props) => {
  const { iconType, text, secondaryText } = props;

  return (
    <div className={cn('icon-text', props)}>
      <Icon mods={{ type: iconType }} mix={['icon-text']} />
      <div className="icon-text__text">{text}</div>
      {secondaryText && (
        <div className="icon-text__text icon-text__secondary-text">{secondaryText}</div>
      )}
    </div>
  );
};

export default IconText;
