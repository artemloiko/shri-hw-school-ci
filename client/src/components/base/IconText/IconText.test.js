import React from 'react';
import { unmountComponentAtNode, render } from 'react-dom';
import { act } from 'react-dom/test-utils';

import IconText from './IconText';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('IconText renders main text', () => {
  const mainText = 'Author';

  act(() => {
    render(<IconText iconType="commit" text={mainText} />, container);
  });
  expect(container.textContent).toBe(mainText);
});

it('IconText renders main and secondary text', () => {
  const mainText = 'Author';
  const secondaryText = 'Commit';

  act(() => {
    render(<IconText iconType="commit" text={mainText} secondaryText={secondaryText} />, container);
  });

  expect(container.textContent).toBe(mainText + secondaryText);
});
