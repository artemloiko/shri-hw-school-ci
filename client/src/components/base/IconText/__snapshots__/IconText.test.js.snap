// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconText with all possible props matches the snapshot 1`] = `
<div
  className="block icon-text block__icon-text"
>
  <div
    className="icon icon_type_commit icon-text__icon"
  />
  <div
    className="icon-text__text"
  >
    master
  </div>
  <div
    className="icon-text__text icon-text__secondary-text"
  >
    90e1126
  </div>
</div>
`;

exports[`IconText with required props matches the snapshot 1`] = `
<div
  className="icon-text"
>
  <div
    className="icon icon_type_commit icon-text__icon"
  />
  <div
    className="icon-text__text"
  >
    master
  </div>
</div>
`;
